Library/Homebrew/cmd/pull.rb

```diff
@@ -113,6 +113,7 @@ def pull
           "git", "diff-tree", "-r", "--name-only",
           "--diff-filter=AM", orig_revision, "HEAD", "--", tap.formula_dir.to_s
         ).each_line do |line|
+          next unless line.end_with? ".rb\n"
```

Comment:
This is a sensible fix, but this fix alone only further masks the bug introduced in 6ba466f5d858c5e6f9d162c11ec03134df6e686e. The following `begin`-`rescue` block is now rather useless and needs to be relocated to the `changed_formulae_names.each do` block a few lines further down. Can you fix that, too? (Skipping non-formula files at this early stage as is done here is still necessary to avoid ending up with bogus stuff in `changed_formulae_names`.)